imports/plugins/core/versions/server/migrations/58_add_reference_id_to_order.js

```diff
@@ -0,0 +1,18 @@
+import { Migrations } from "meteor/percolate:migrations";
+import * as collections from "/lib/collections";
+
+
+Migrations.add({
+  version: 58,
+  up() {
+    // In case any orders don't have a reference ID
+    collections.Orders.find({ "referenceId" : { "$exists" : false }}).forEach((order) => {
+      collections.Orders.findOneAndUpdate({ _id: order._id }, { $set: { referenceId: order._id }})
+    });
+
+    // In case we have existing carts
+    collections.Cart.find({ "referenceId" : { "$exists" : false }}).forEach((cart) => {
+      collections.Cart.findOneAndUpdate({ _id: cart._id }, { $set: { referenceId: cart._id }})
+    })
```

Comment:
It's best to use our `findAndConvertInBatches` util function since these collections can be large. This was already done for orders in migration 47. I would do only cart here, which means you could simply copy the `47_order_ref.js` file contents into here and change `Orders` to `Cart` and `order` to `cart` everywhere.